sound/soc/intel/boards/sof_sdw.c

```diff
@@ -937,7 +937,7 @@ static int mc_probe(struct platform_device *pdev)
 		amp_num += codec_info_list[i].amp_num;
 
 	card->components = devm_kasprintf(card->dev, GFP_KERNEL,
-					  "cfg-spk:%d, cfg-amp:%d",
+					  "cfg-spk:%d cfg-amp:%d",
 					  (sof_sdw_quirk & SOF_SDW_FOUR_SPK)
 					  ? 4 : 2, amp_num);
```

Comment:
@plbossart  The error found by perexg is caused by SKU:0990 . Can we just set spk number based on amp_num which is the real number of codec, so we don't need sof_sdw_quirk for spk number.  Do we need spk number or exact amp number ?

Reply:
I think we need both.

The number of amps is needed to figure out the configuration at the driver level. The number of amps is inferred from the ACPI configuration.
The number of speakers is needed for UCM to program the controls in each amp. We have no mechanism to describe the speakers in ACPI so we will need a DMI quirk for this case.

Reply:
thanks, I got it.